### ISSUE TITLE
Add xonsh shell support

### ISSUE THREAD

**Issue by anki-code** [withfig/fig#69]
Hi! Thank you for the interesting project!
I'm going to create this by reference from Matt Schrage.

Please add [the xonsh shell](https://xon.sh/) support.

Thanks!

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by mattschrage** [comment]
Thanks for creating the issue!

**Comment by mattschrage** [comment]
Hey @anki-code! 

Briefly looked into this just now and it seems like seems like adding `xonsh` support is possible!

We would need to add Fig hooks to the `on_precommand` and `on_post_prompt ` [events](https://xon.sh/events.html). You can see the `zsh` [implementation](https://github.com/withfig/config/blob/main/fig.zsh) for reference. 

All of the shell integrations are [open source](https://github.com/withfig/config), so if you're interested in contributing this, let me know! 

(PS. Do you still need an invite?)

**Comment by anki-code** [comment]
> Briefly looked into this just now and it seems like seems like adding xonsh support is possible!

Yeah, with xonsh you can implement significant more incredible things with Fig.

> We would need to add Fig hooks to the on_precommand and on_post_prompt events. 

Also this could be implemented as a xontrib (xonsh contribution/extension) for Fig:
* Examples of [xontribs on Github](https://github.com/topics/xontrib)
* [Cookiecutter template](https://github.com/xonsh/xontrib-cookiecutter) to create a xontrib in 10 minutes
* [xontrib tutorial](https://xon.sh/tutorial_xontrib.html)

> All of the shell integrations are open source, so if you're interested in contributing this, let me know!
> (PS. Do you still need an invite?)

It will be great if you make a first try and put here a few words about result. Then I can give an advice or continue the creation of proof of concept when I'll back to Fig.

**Comment by KadenWolff** [comment]
Hey, I'm interested in working on this integration. Could someone direct me to some resources to start building this, and any advice for doing this? Thanks!
